crates/cache/src/config.rs

```diff
@@ -579,6 +586,187 @@ impl CacheConfig {
     }
 }
 
+/// A builder for `CacheConfig`s.
+#[derive(Debug, Clone, Default)]
+pub struct CacheConfigBuilder {
+    enabled: bool,
+    directory: Option<PathBuf>,
+    worker_event_queue_size: Option<u64>,
+    baseline_compression_level: Option<i32>,
+    optimized_compression_level: Option<i32>,
+    optimized_compression_usage_counter_threshold: Option<u64>,
+    cleanup_interval: Option<Duration>,
+    optimizing_compression_task_timeout: Option<Duration>,
+    allowed_clock_drift_for_files_from_future: Option<Duration>,
+    file_count_soft_limit: Option<u64>,
+    files_total_size_soft_limit: Option<u64>,
+    file_count_limit_percent_if_deleting: Option<u8>,
+    files_total_size_limit_percent_if_deleting: Option<u8>,
+}
+
+impl CacheConfigBuilder {
+    /// Specifies whether the cache system is used or not.
+    pub fn enabled(mut self, enabled: bool) -> Self {
+        self.enabled = enabled;
+        self
+    }
+
+    /// Specifies where the cache directory is. Must be an absolute path.
+    pub fn directory(mut self, directory: impl Into<PathBuf>) -> Self {
```

Comment:
While we don't have hard-and-fast conventions in Wasmtime I at least personally prefer builders to take `&mut self` and return `&mut Self` (like `Config` methods). Would you be up to changing to that pattern?

Reply:
I had a feeling this one might come up, and I figured it was a 50/50 chance I got it right :) 
Not a problem